test/integration/Client/client_lib_thread_interdependency_test.cpp

```diff
@@ -0,0 +1,499 @@
+    }
+}
+
+void update_chronicle_threads_state(int tid, ThreadState new_state)
+{
+    for(size_t i = 0; i < shared_state.size(); ++i)
+    {
+        if(i == tid) continue;
+        if((i % 2 == tid % 2))
+        {
+            shared_state[i] = static_cast<int>(new_state);
+            LOG_INFO("[ClientLibThreadInterdependencyTest] -Thread {}- Thread {} state updated to: {}", tid, i
+                     , get_state_name(new_state));
+        }
+    }
+}
+
+
+void update_story_threads_state(int tid, ThreadState new_state)
+{
+    for(size_t i = 0; i < shared_state.size(); ++i)
+    {
+        if(i == tid) continue;
+        if((i % 2 == tid % 2) && ((i % 4) / 2 == (tid % 4) / 2))
+        {
+            shared_state[i] = static_cast<int>(new_state);  // Update state
+            LOG_INFO("[ClientLibThreadInterdependencyTest] -Thread {}- Thread {} state updated to: {}", tid, i
+                     , get_state_name(new_state));
+        }
+    }
+}
+
+
+void check_thread_initialization(int tid, int ret)
+{
+    std::lock_guard <std::mutex> lock(shared_state_mutex);
+    if(static_cast<ThreadState>(shared_state[tid]) == ThreadState::UNKNOWN)
+    {
+        shared_state[tid] = static_cast<int>(ThreadState::THREAD_INITIALIZED);
+        LOG_INFO("[ClientLibThreadInterdependencyTest] -Thread {}- State updated to: {}", tid, get_state_name(
+                ThreadState::THREAD_INITIALIZED));
+    }
+}
+
+void check_chronicle_created(int tid, int ret)
+{
+    std::lock_guard <std::mutex> lock(shared_state_mutex);
+    if(ret == chronolog::CL_SUCCESS)
+    {
+        if(static_cast<ThreadState>(shared_state[tid]) == ThreadState::THREAD_INITIALIZED)
+        {
+            shared_state[tid] = static_cast<int>(ThreadState::CHRONICLE_CREATED);
+            LOG_INFO("[ClientLibThreadInterdependencyTest] -Thread {}- State updated to: {}", tid, get_state_name(
+                    ThreadState::CHRONICLE_CREATED));
+            update_chronicle_threads_state(tid, ThreadState::CHRONICLE_CREATED);
+        }
+        else
+        {
+            LOG_ERROR("[ClientLibThreadInterdependencyTest] -Thread {}- received a CL_SUCCESS when transitioning to "
+                      "CHRONICLE_CREATED state from a state different than THREAD_INITIALIZED", tid);
+        }
+    }
+    else if(ret == chronolog::CL_ERR_CHRONICLE_EXISTS)
+    {
+        if(static_cast<ThreadState>(shared_state[tid]) != ThreadState::UNKNOWN ||
+           static_cast<ThreadState>(shared_state[tid]) != ThreadState::THREAD_INITIALIZED)
+        {
+            LOG_INFO("[ClientLibThreadInterdependencyTest] -Thread {}- received a CL_ERR_CHRONICLE_EXISTS return "
+                     "value when trying to create a chronicle on a state different from UNKNOWN or THREAD_INITIALIZED"
+                     , tid);
+        }
+        else
+        {
+            LOG_ERROR("[ClientLibThreadInterdependencyTest] -Thread {}- received a CL_ERR_CHRONICLE_EXISTS return value"
```

Comment:
Not reachable. `shared_state[tid]` cannot be `ThreadState::UNKNOWN` and `ThreadState::THREAD_INITIALIZED` to make the if-statement fail.